wayland-scanner/src/interfaces.rs

```diff
@@ -0,0 +1,106 @@
+use proc_macro2::{Ident, Span, TokenStream};
+
+use wayland_commons::scanner::{Interface, Message, Protocol, Type};
+
+use quote::quote;
+
+pub fn generate(protocol: &Protocol) -> TokenStream {
+    let interfaces = protocol.interfaces.iter().map(generate_interface);
+    quote!( #(#interfaces)* )
+}
+
+fn generate_interface(interface: &Interface) -> TokenStream {
+    let const_name = Ident::new(
+        &format!("{}_INTERFACE", interface.name.to_ascii_uppercase()),
+        Span::call_site(),
+    );
```

Comment:
Quote contains [`format_ident!`](https://docs.rs/quote/1.0.9/quote/macro.format_ident.html) to simplify the pattern of `Ident::new(&format!("..."), ...)`.